db/migrate/20151119070640_remove_dangling_entries.rb

```diff
@@ -0,0 +1,22 @@
+class RemoveDanglingEntries < ActiveRecord::Migration
+  def up
+    UserSnp.find_each do |us|
+      if Snp.find_by_name(us.snp_name).nil? || Genotype.find_by_id(us.id).nil?
+        UserSnp.destroy(us.id)
+      end
+    end
```

Comment:
Let me check whether we even have userSNPs without genotypes - if we don't have any, then we don't have to bother to delete them. It may be a bit of a hassle to find out why a specific dangling user-SNP is dangling in the first place, whether we're deleting data that a user has requested to delete and that hasn't been deleted for some reason (crash), or whether we're deleting data that a user has added to the site but that hasn't been properly linked (crash)